### PR TITLE
climate vulnerability section

### PR DESCRIPTION
**Pull request by Bluesmile82**
This is a PR for the Climate vulnerability section a.k.a poverty-country section
- [x] Create new section in country page
- [x] Connect component to redux
- [x] Create adaptations provider
- [x] Add sections to the component
- [x] Create graphs for the widgets
- [x] Make all the info buttons work

### CODE REVIEW

File: app/javascript/app/components/country-climate-vulnerability/country-climate-vulnerability-styles.scss [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  color: $theme-color;
+  display: flex;
+  align-items: center;
+}
+
+.info {
+  flex-basis: 100%;
+}
+
+.infoText {
```
Comment (j8seangel) [inline app/javascript/app/components/country-climate-vulnerability/country-climate-vulnerability-styles.scss]: We are not using this selector, aren't we?

File: app/javascript/app/components/country-climate-vulnerability/country-climate-vulnerability-selectors.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  );
+};
+
+const getListData = (sections, iso, slug) => {
+  const items = sections.filter(section => section.slug.startsWith(slug));
+  const itemList = [];
+  if (
+    !items.some(item => item.values.some(country => country.location === iso))
+  ) {
+    itemList.push('No data');
```
Comment (j8seangel) [inline app/javascript/app/components/country-climate-vulnerability/country-climate-vulnerability-selectors.js]: Why don't return the empty array directly?

File: app/javascript/app/actions.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import { actions as countriesProvider } from 'providers/countries-provider';
 import { actions as regionsProvider } from 'providers/regions-provider';
 import { actions as ndcsSdgsMetaProvider } from 'providers/ndcs-sdgs-meta-provider';
+import { actions as adaptationsProvider } from 'providers/adaptations-provider';
```
Comment (j8seangel) [inline app/javascript/app/actions.js]: Please update to use the new syntax so we don't need this file anymore.

File: app/javascript/app/components/info-icon/info-icon-component.jsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        })}
+        onClick={toggleInfo}
+      >
+        <Icon icon={infoIcon} />
+      </Button>
+    );
+  }
+}
+
+InfoIcon.propTypes = {
```
Comment (j8seangel) [inline app/javascript/app/components/info-icon/info-icon-component.jsx]: We don't need all of this props as we could pass our custom className to customize the styles.

File: app/javascript/app/components/country-climate-vulnerability/country-climate-vulnerability-component.jsx [hunk 5]
```diff
@@ -0,0 +1,98 @@
+import React, { PureComponent } from 'react';
+import Proptypes from 'prop-types';
+import cx from 'classnames';
+
+import PrepLink from 'components/prep-link';
+import Button from 'components/button';
+import Card from 'components/card';
+import Icon from 'components/icon';
+import VulnerabilityGraph from 'components/vulnerability-graph';
+import infoIcon from 'assets/icons/info.svg';
```
Comment (j8seangel) [inline app/javascript/app/components/country-climate-vulnerability/country-climate-vulnerability-component.jsx]: I like the idea of having the info icon only if we use it

File: app/javascript/app/components/prep-link/prep-link-component.jsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import React, { PureComponent } from 'react';
+import prepLogo from 'assets/icons/prep-logo.svg';
+import Icon from 'components/icon';
+import styles from './prep-link-styles.scss';
+
+class PrepLink extends PureComponent {
```
Comment (j8seangel) [inline app/javascript/app/components/prep-link/prep-link-component.jsx]: I don't like the idea of creating a component just for that... at least we could create a component that accepts the image and content as props or include it in the poverty folder.
But having a general component just for this I don't think is very useful.

File: app/javascript/app/components/info-icon/info-icon-component.jsx [hunk 7]
```diff
@@ -0,0 +1,35 @@
+import React, { PureComponent } from 'react';
```
Comment (j8seangel) [inline app/javascript/app/components/info-icon/info-icon-component.jsx]: Could we move this component to the buttons folder? There are so many components in the root that is hard to find what you are looking for.

File: app/javascript/app/components/vulnerability-graph/vulnerability-graph-component.jsx [hunk 8]
```diff
@@ -0,0 +1,109 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import CircularChart from 'components/circular-chart';
+import InfoIcon from 'components/info-icon';
+import { COUNTRIES_NUMBER } from 'data/country-codes';
```
Comment (j8seangel) [inline app/javascript/app/components/vulnerability-graph/vulnerability-graph-component.jsx]: I'd prefer to calculate this number dynamically with the number of countries in the request.
Is that possible? Maybe in the action ?

File: app/javascript/app/components/vulnerability-graph/vulnerability-graph-component.jsx [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+    ) : (
+      'No data'
+    );
+  }
+
+  renderLine() {
+    const { sectionData } = this.props;
+    const value =
+      sectionData.data && Math.floor(sectionData.data.value * 10) / 10;
+    const riskValue = parseInt(sectionData.data.rank.absolute, 10) * 100;
```
Comment (j8seangel) [inline app/javascript/app/components/vulnerability-graph/vulnerability-graph-component.jsx]: When we have more than 100% the line is out of the graph, could we add a limit a 100% limit?
image

File: app/javascript/app/components/country-climate-vulnerability/country-climate-vulnerability-selectors.js (hunk 2 above)
Comment (Bluesmile82) [reply app/javascript/app/components/country-climate-vulnerability/country-climate-vulnerability-selectors.js]: It is maybe different having no vulnerabilities or having no data about a country. We will have to check the final data.

File: app/javascript/app/components/vulnerability-graph/vulnerability-graph-component.jsx (hunk 9 above)
Comment (Bluesmile82) [reply app/javascript/app/components/vulnerability-graph/vulnerability-graph-component.jsx]: We shouldn't have more than 100% 😮  That's pretty risky

File: app/javascript/app/components/vulnerability-graph/vulnerability-graph-component.jsx [hunk 10]
```diff
@@ -69,7 +69,7 @@ class VulnerabilityGraph extends PureComponent {
         </div>
       </div>
     ) : (
-      'No data'
+      <div className={styles.noData}>No data</div>
```
Comment (j8seangel) [inline app/javascript/app/components/vulnerability-graph/vulnerability-graph-component.jsx]: We already have a component called `NoContent` for that kind of this, could you please use it?

**Pull request merged**